### PR TITLE
Allow a user to add or override deserialize configuration in karaf

### PR DESCRIPTION
**Pull request by tbouron**
This adds:
- an empty file for the user to edit
- an integration tests (although, the entire test suite does not work, the setup has some issue that needs to be fixed)

### CODE REVIEW

**Comment by aledsage** [comment]
This depends on https://github.com/apache/brooklyn-server/pull/492 being merged first, I presume.

File: karaf/apache-brooklyn/src/main/resources/etc/org.apache.brooklyn.classrename.cfg [hunk 1]
```diff
@@ -0,0 +1,21 @@
+################################################################################
```
Comment (m4rkmckenna) [inline karaf/apache-brooklyn/src/main/resources/etc/org.apache.brooklyn.classrename.cfg]: This file should probably be added via the brooklyn `feature.xml` also to make extension / upgrade by downstream projects easier
```
<feature name="example-config" version="${project.version}">
        <configfile finalname="${karaf.etc}/${groupId}.classrename.cfg" override="true">
            mvn:${project.groupId}/${project.artifactId}/${project.version}/cfg/classrename
        </configfile>
</feature>
```

Comment (tbouron) [inline karaf/apache-brooklyn/src/main/resources/etc/org.apache.brooklyn.classrename.cfg]: @m4rkmckenna Do you mean that I shouldn't add it manually there but just add a feature on the `feature.xml`? Is it a pattern well known? It seems we currently do not use that

**Comment by m4rkmckenna** [comment]
@tbouron The pattern is probably not that well known but IMO it's cleaner and the cfg files are versioned

File: karaf/apache-brooklyn/src/main/resources/etc/org.apache.brooklyn.classrename.cfg (hunk 1 above)
Comment (m4rkmckenna) [inline karaf/apache-brooklyn/src/main/resources/etc/org.apache.brooklyn.classrename.cfg]: @tbouron On second thoughts lets not hold up this PR changing the feature as there are other config files that do not follow this pattern 

@svet What do you think about creating a common maven artifact that contains all the additional brooklyn config so downstream projects can include them easily

eg `default.catalog.bom`, `org.apache.brooklyn.classrename.cfg`, `org.apache.brooklyn.core.catalog.bomscanner.cfg`, `org.apache.brooklyn.osgilauncher.cfg`

Comment (neykov) [inline karaf/apache-brooklyn/src/main/resources/etc/org.apache.brooklyn.classrename.cfg]: I believe there's some mechanism to have the files self-extract on first boot or even during assembly. Should look it up.

I agree we should make the distribution assembly re-usable, not only files in etc.

Comment (tbouron) [inline karaf/apache-brooklyn/src/main/resources/etc/org.apache.brooklyn.classrename.cfg]: @neykov @m4rkmckenna I created https://github.com/apache/brooklyn-server/pull/504 in `brooklyn-server` and https://github.com/apache/brooklyn-dist/pull/75 here to do this.

The current implementation will self-extract the files on first boot

**Comment by tbouron** [comment]
@aledsage This can be merged now.

Also, I created 2 PRs to follow up, based on @m4rkmckenna and @neykov's comments

1. https://github.com/apache/brooklyn-server/pull/504
2. #75

**Comment by geomacy** [comment]
I tried testing out a custom bundle and couldn't get it to rebind using the user specific file introduced here.  See my test files and procedure at https://github.com/geomacy/brooklyn-dist/pull/1.  I'm assuming I'm doing something wrong but worth recording it here.

**Comment by geomacy** [comment]
Also a variant that just changed the package name had the same problem.

**Pull request merged**